msctl

```diff
@@ -346,9 +351,9 @@ getJavaMemory() {
 serverRunning() {
   # Try to determine if the world is running.
   if [ $(getJavaPID "$1") -gt 0 ]; then
-    return 0
+    echo 0
   else
-    return 1
+    echo 1
```

Comment:
I don't think this change is necessary. It forces a lot of other changes that make the code less readable.

E.g.
```
-  while serverRunning $1; do
+  while [ $(serverRunning $1) -eq 0 ]; do
```

Reply:
Usually 0 means false, and 1 means true. UNIX land returns 0 when a command successfully ran and a value >0 for error states. This return trick was used here to make the code more readable, but the values are inverted from what you mean when changed to echo.

Reply:
Thanks for clarifying this. It's confusing because, as you said, in UNIX land 0 is returned when the command ran successfully. In my head, I associate success with being "TRUE", so it makes me think it should return 1 based on the 0=false and 1=true convention, but it doesn't.... But anyway, I think I understand now, thanks

Reply:
We should probably add comments to the serverRunning method to note this UNIX oddity.

```patch
serverRunning() {
  # Try to determine if the world is running.
  if [ $(getJavaPID "$1") -gt 0 ]; then
-    return 0
+    return 0  # Return a true value.
  else
-    return 1
+    return 1  # Return a false value.
  fi
}

```

Reply:
@sandain do you know why `if [serverRunning $1]` doesn't work? it only works when i omit the brackets and I can't figure out how to do it if I want to negate the expression

Reply:
You need spaces next to the square brackets.
```
if [ serverRunning $1 ]
```

to negate:
```
if [ ! serverRunning $1 ]
```

Reply:
Although, I think the brackets are optional.

```
if ! serverRunning $1; then
```